src/validate.ts

```diff
@@ -1,14 +1,15 @@
 import { readOperationsRepository } from './utils/filesystem';
 import { runAndHandleErrors } from './utils/cli';
 import { validate } from './utils/validation';
+import { replacer } from './utils/marshaling';
 
 const main = async () => {
   const rawOpsData = readOperationsRepository();
 
   const [success, logs] = validate(rawOpsData);
   if (!success) {
     console.log('Validation failed:');
-    console.log(JSON.stringify(logs, null, 2));
+    console.log(JSON.stringify(logs, replacer, 2));
```

Comment:
Is the `logs` variable really logs? IIUC it is the result of safeParse from zod. Does this work? (e.g. print out the zod violations)

Reply:
When parsing succeeds `logs` is the pass-through data.
When parsing fails `logs` are the error messages. Below is an example:
(btw, well done on trail blazing with Zod, it's cool)
```
Validation failed:
{
  "error": {
    "addIssue": {},
    "addIssues": {},
    "format": {},
    "issues": [
      {
        "code": "invalid_type",
        "expected": "number",
        "message": "Required",
        "path": [
          "apis",
          "api3",
          "deployments",
          "2022-03-22",
          "config",
          "chains",
          0,
          "maxConcurrency"
        ],
        "received": "undefined"
      }
    ],
    "name": "ZodError"
  },
  "success": false
}

```